src/main/webapp/app/exam/manage/exams/exam-exercise-import/exam-exercise-import.component.ts

```diff
@@ -78,14 +85,21 @@ export class ExamExerciseImportComponent implements OnInit {
         this.exam.exerciseGroups?.forEach((exerciseGroup) => {
             this.selectedExercises.set(exerciseGroup, new Set<Exercise>(exerciseGroup.exercises));
         });
+        const duplicated = new Set<string>();
         // Initialize containsProgrammingExercises
         this.exam.exerciseGroups!.forEach((exerciseGroup) => {
             const hasProgrammingExercises = !!exerciseGroup.exercises?.some((value) => value.type === ExerciseType.PROGRAMMING);
             this.containsProgrammingExercises.set(exerciseGroup, hasProgrammingExercises);
             // In case of a rejected import, we can delete programming exercises with a title from the Map / blocklist, as those were not rejected by the server.
             exerciseGroup.exercises?.forEach((exercise) => {
                 if (exercise.type === ExerciseType.PROGRAMMING && exercise.title) {
-                    this.titleAndShortNameOfProgrammingExercises.delete(exercise.id!);
+                    if (!duplicated.has(exercise.title)) {
+                        duplicated.add(exercise.title);
+                        this.titleAndShortNameOfProgrammingExercises.delete(exercise.id!);
+                    } else {
+                        this.titleAndShortNameOfProgrammingExercises.set(exercise.id!, [exercise.title!, '']);
+                        exercise.title = '';
+                    }
```

Comment:
```suggestion
                    if (duplicated.has(exercise.title)) {
                        this.titleAndShortNameOfProgrammingExercises.set(exercise.id!, [exercise.title!, '']);
                        exercise.title = '';
                    } else {
                        duplicated.add(exercise.title);
                        this.titleAndShortNameOfProgrammingExercises.delete(exercise.id!);
                    }
```

Is the condition not easier to follow the other way around?